src/gui/search/searchtab.cpp

```diff
@@ -154,19 +174,131 @@ QStandardItemModel* SearchTab::getCurrentSearchListModel() const
+        sizeInBytes(maxSize->value(), static_cast<SizeUnit>(maxSizeUnit->currentIndex())));
+    filterModel->invalidate();
+    updateResultsCount();
+}
+
+void SearchTab::fillFilterComboBoxes()
+{
+    using Utils::Misc::SizeUnit;
+    QStringList unitStrings;
+    unitStrings.append(unitString(SizeUnit::Byte));
+    unitStrings.append(unitString(SizeUnit::KibiByte));
+    unitStrings.append(unitString(SizeUnit::MebiByte));
+    unitStrings.append(unitString(SizeUnit::GibiByte));
+    unitStrings.append(unitString(SizeUnit::TebiByte));
+    unitStrings.append(unitString(SizeUnit::PebiByte));
+    unitStrings.append(unitString(SizeUnit::ExbiByte));
+
+    minSizeUnit->clear();
+    maxSizeUnit->clear();
+    minSizeUnit->addItems(unitStrings);
+    maxSizeUnit->addItems(unitStrings);
+
+    minSize->setValue(0);
+    minSizeUnit->setCurrentIndex(static_cast<int>(SizeUnit::MebiByte));
+
+    maxSize->setValue(-1);
+    maxSizeUnit->setCurrentIndex(static_cast<int>(SizeUnit::TebiByte));
+
+    filterMode->clear();
+
+    QMetaEnum nameFilteringModeEnum =
+        this->metaObject()->enumerator(this->metaObject()->indexOfEnumerator("NameFilteringMode"));
+
+    filterMode->addItem(tr("Torrent names only"), nameFilteringModeEnum.valueToKey(OnlyNames));
+    filterMode->addItem(tr("Everywhere"), nameFilteringModeEnum.valueToKey(Everywhere));
+
+    QVariant selectedMode = SettingsStorage::instance()->loadValue(QLatin1String(FILTER_MODE_SETTING_NAME),
+                                                                   nameFilteringModeEnum.valueToKey(OnlyNames));
+    int index = filterMode->findData(selectedMode);
+    filterMode->setCurrentIndex(index == -1 ? 0 : index);
+}
+
+QString SearchTab::statusText(SearchTab::Status st)
+{
+    switch (st) {
+    case Status::Ongoing:
+        return tr("Searching...");
+    case Status::Finished:
+        return tr("Search has finished");
+    case Status::Aborted:
+        return tr("Search aborted");
+    case Status::Error:
+        return tr("An error occurred during search...");
+    case Status::NoResults:
+        return tr("Search returned no results");
+    default:
+        return QString();
+    }
+}
+
+QString SearchTab::statusIconName(SearchTab::Status st)
+{
+    switch (st) {
+    case Status::Ongoing:
+        return QLatin1String("task-ongoing");
+    case Status::Finished:
+        return QLatin1String("task-complete");
+    case Status::Aborted:
+        return QLatin1String("task-reject");
+    case Status::Error:
+        return QLatin1String("task-attention");
+    case Status::NoResults:
+        return QLatin1String("task-attention");
+    default:
+        return QString();
+    }
+}
+
+void SearchTab::saveFilterModeAsDefault()
+{
+    SettingsStorage::instance()->storeValue(QLatin1String(FILTER_MODE_SETTING_NAME),
+                                            filterMode->itemData(filterMode->currentIndex()));
```

Comment:
I assume if you elect to save/load the other values too, it makes sense to do it inside updateFilter() instead.